Src/Support/Google.Apis.Auth.Tests/OAuth2/ExternalAccountCredentialTestsBase.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            Assert.Equal(TokenUrl, accessTokenRequest.RequestUri.ToString());
+            Assert.Equal(HttpMethod.Post, accessTokenRequest.Method);
+
+            string contentText = WebUtility.UrlDecode(await accessTokenRequest.Content.ReadAsStringAsync());
+
+            if (isWorkforce)
+            {
+                Assert.Null(accessTokenRequest.Headers.Authorization);
+                Assert.Contains($"options={{\"userProject\":\"{WorkforcePoolUserProject}\"}}", contentText);
+            }
+            else
+            {
+                Assert.Equal("Basic", accessTokenRequest.Headers.Authorization.Scheme);
+                Assert.Equal(Convert.ToBase64String(Encoding.UTF8.GetBytes($"{ClientId}:{ClientSecret}")), accessTokenRequest.Headers.Authorization.Parameter);
+                Assert.DoesNotContain("options=", contentText);
+            }
+
+            Assert.Contains(GrantTypeClaim, contentText);
+            Assert.Contains(RequestedTokenTypeClaim, contentText);
+            Assert.Contains($"audience={Audience}", contentText);
+            Assert.Contains($"subject_token_type={SubjectTokenType}", contentText);
+            Assert.Contains($"subject_token={SubjectTokenText}", contentText);
+            Assert.Contains($"scope={scope}", contentText);
+
+            return new HttpResponseMessage
+            {
+                Content = new StringContent(
+                    NewtonsoftJsonSerializer.Instance.Serialize(new TokenResponse
+                    {
+                        AccessToken = AccessToken,
+                        ExpiresInSeconds = 24 * 60 * 60,
+                    }), Encoding.UTF8)
+            };
+        }
+
+        protected static async Task<HttpResponseMessage> ValidateImpersonatedAccessTokenRequest(HttpRequestMessage accessTokenRequest)
+        {
+            Assert.Equal(ImpersonationUrl, accessTokenRequest.RequestUri.ToString());
+            Assert.Equal(HttpMethod.Post, accessTokenRequest.Method);
+
+            Assert.Contains(accessTokenRequest.Headers, header => header.Key == QuotaProjectHeaderName && header.Value.Single() == QuotaProject);
+
+            Assert.Equal("Bearer", accessTokenRequest.Headers.Authorization.Scheme);
+            Assert.Equal(AccessToken, accessTokenRequest.Headers.Authorization.Parameter);
+
+            string contentText = WebUtility.UrlDecode(await accessTokenRequest.Content.ReadAsStringAsync());
+
+            Assert.Contains(Scope, contentText);
+
+            return new HttpResponseMessage()
+            {
+                Content = new StringContent(
+                    NewtonsoftJsonSerializer.Instance.Serialize(new
+                    {
+                        accessToken = ImpersonatedAccessToken,
+                        expireTime = "2020-05-13T16:00:00.045123456Z"
+                    }))
```

Comment:
This one doesn't specify UTF-8 - is that relevant?

Reply:
After extracting it does, not sure if it was default already.